bazel/grpc_deps.bzl

```diff
@@ -445,6 +445,17 @@ def grpc_deps():
             ],
         )
 
+    if "com_github_cncf_udpa" not in native.existing_rules():
+        http_archive(
+            name = "com_github_cncf_udpa",
+            sha256 = "5bc8365613fe2f8ce6cc33959b7667b13b7fe56cb9d16ba740c06e1a7c4242fc",
+            strip_prefix = "xds-cb28da3451f158a947dfc45090fe92b07b243bc1",
+            urls = [
+                "https://storage.googleapis.com/grpc-bazel-mirror/github.com/cncf/xds/archive/cb28da3451f158a947dfc45090fe92b07b243bc1.tar.gz",
+                "https://github.com/cncf/xds/archive/cb28da3451f158a947dfc45090fe92b07b243bc1.tar.gz",
+            ],
+        )
```

Comment:
Do we need to add UDPA as a git submodule? How will this PR impact CMake builds?

Reply:
Since non-bazel builds use the existing pre-generated upb files, they won't be affected by this change. And this udpa is a submodule of gRPC. (third_party/xds) FYI, I chose `com_github_cncf_udpa` to have the same name as Envoy.

Reply:
Good to know.

Reply:
It should actually be called `com_github_cncf_xds`, not `com_github_cncf_udpa`.  The latter name is deprecated.

Reply:
This needs to be the same with https://github.com/envoyproxy/data-plane-api/blob/4140ad93e190cf4aa6a35b4caac1776fa640953e/bazel/repository_locations.bzl#L42 so that they're resolved to the same target.

Reply:
That's unfortunate.

@htuch, how hard would it be to rename this target in the Envoy repo?  It seems like these dependency names are basically part of the API used by other projects, so we should probably remove the term "udpa" from them.

Reply:
I can rename this in line with Envoy's one if Envoy wants to change it. And this can be done later so I'm going to merge this to unblock other changes relying on this.

Reply:
Okay, I guess that works.  Thanks.

Reply:
We can rename this internally, please go ahead, thanks.